gum/backend-posix/gumexceptor-posix.c

```diff
@@ -378,6 +396,236 @@ gum_exceptor_backend_on_signal (int sig,
+    case X86_INS_MOVSX:
+    case X86_INS_MOVSXD:
+    case X86_INS_MOVSW:
+    case X86_INS_MOVD:
+    case X86_INS_MOVQ:
+    case X86_INS_MOVDQ2Q:
+    case X86_INS_RDRAND:
+    case X86_INS_RDSEED:
+    case X86_INS_RDMSR:
+    case X86_INS_RDPMC:
+    case X86_INS_RDTSC:
+    case X86_INS_RDTSCP:
+    case X86_INS_CRC32:
+    case X86_INS_SHA1MSG1:
+    case X86_INS_SHA1MSG2:
+    case X86_INS_SHA1NEXTE:
+    case X86_INS_SHA1RNDS4:
+    case X86_INS_SHA256MSG1:
+    case X86_INS_SHA256MSG2:
+    case X86_INS_SHA256RNDS2:
+    case X86_INS_AESDECLAST:
+    case X86_INS_AESDEC:
+    case X86_INS_AESENCLAST:
+    case X86_INS_AESENC:
+    case X86_INS_AESIMC:
+    case X86_INS_AESKEYGENASSIST:
+    case X86_INS_PACKSSDW:
+    case X86_INS_PACKSSWB:
+    case X86_INS_PACKUSWB:
+    case X86_INS_XCHG:
+    case X86_INS_CLD:
+    case X86_INS_STD:
+      switch (X86OP (0).type)
+      {
+        case X86_OP_MEM:
+          return GUM_MEMOP_WRITE;
+        case X86_OP_REG:
+          if (X86OP (1).type == X86_OP_MEM)
+            return GUM_MEMOP_READ;
+        default:
+          return GUM_MEMOP_READ;
+      }
+    default:
+      return GUM_MEMOP_READ;
+  }
+}
+
+#elif defined (HAVE_ARM64)
+
+static GumMemoryOperation
+gum_exceptor_arm64_check_mem_operation (cs_insn * insn)
+{
+  switch (insn->id)
+  {
+    case ARM64_INS_LDUR:
+    case ARM64_INS_LDURB:
+    case ARM64_INS_LDRSW:
+    case ARM64_INS_LDRSB:
+    case ARM64_INS_LDRSH:
+    case ARM64_INS_LDR:
+    case ARM64_INS_LDURSW:
+    case ARM64_INS_LDP:
+    case ARM64_INS_LDNP:
+    case ARM64_INS_LDPSW:
+    case ARM64_INS_LDRH:
+    case ARM64_INS_LDRB:
+    case ARM64_INS_LDRAA:
+    case ARM64_INS_LDRAB:
+      return GUM_MEMOP_READ;
+    case ARM64_INS_STRB:
+    case ARM64_INS_STURB:
+    case ARM64_INS_STUR:
+    case ARM64_INS_STR:
+    case ARM64_INS_STP:
+    case ARM64_INS_STNP:
+    case ARM64_INS_STXR:
+    case ARM64_INS_STXRH:
+    case ARM64_INS_STLXRH:
+    case ARM64_INS_STXRB:
+      return GUM_MEMOP_WRITE;
+    default:
+      return GUM_MEMOP_READ;
+  }
+}
+
+#elif defined (HAVE_ARM)
+
+static GumMemoryOperation
+gum_exceptor_arm_check_mem_operation (cs_insn * insn)
+{
+  switch (insn->id)
+  {
+    case ARM_INS_LDREX:
+    case ARM_INS_LDREXB:
+    case ARM_INS_LDREXD:
+    case ARM_INS_LDREXH:
+    case ARM_INS_LDR:
+    case ARM_INS_LDRD:
+    case ARM_INS_LDRB:
+    case ARM_INS_LDRBT:
+    case ARM_INS_LDRH:
+    case ARM_INS_LDRHT:
+    case ARM_INS_LDRSB:
+    case ARM_INS_LDRSBT:
+    case ARM_INS_LDRSH:
+    case ARM_INS_LDRSHT:
+    case ARM_INS_LDRT:
+      return GUM_MEMOP_READ;
+    case ARM_INS_STREX:
+    case ARM_INS_STREXB:
+    case ARM_INS_STREXD:
+    case ARM_INS_STREXH:
+    case ARM_INS_STR:
+    case ARM_INS_STRB:
+    case ARM_INS_STRD:
+    case ARM_INS_STRBT:
+    case ARM_INS_STRH:
+    case ARM_INS_STRHT:
+    case ARM_INS_STRT:
+      return GUM_MEMOP_WRITE;
+    default:
+      return GUM_MEMOP_READ;
+  }
+}
+
+#endif
```

Comment:
For consistency, ordering of architecture-specific code should be:
- x86/x86_64
- arm
- arm64
- mips